connect/api/src/main/java/org/apache/kafka/connect/sink/SinkRecord.java

```diff
@@ -79,17 +195,21 @@ public boolean equals(Object o) {
 
         SinkRecord that = (SinkRecord) o;
 
-        if (kafkaOffset != that.kafkaOffset)
-            return false;
-
-        return timestampType == that.timestampType;
+        return kafkaOffset == that.kafkaOffset &&
+                timestampType == that.timestampType &&
+                Objects.equals(originalTopic, that.originalTopic) &&
+                Objects.equals(originalKafkaPartition, that.originalKafkaPartition)
+                && originalKafkaOffset == that.originalKafkaOffset;
```

Comment:
Nit: consistent formatting
```suggestion
                Objects.equals(originalKafkaPartition, that.originalKafkaPartition) &&
                originalKafkaOffset == that.originalKafkaOffset;
```